tests/test_register_schema.py

```diff
@@ -0,0 +1,130 @@
+from contextlib import redirect_stderr
+import json
+import jsonschema
+import logging
+import pytest
+import io
+
+from jupyter_telemetry.eventlog import EventLog
+
+
+def test_register_invalid():
+    """
+    Test registering invalid schemas fails
+    """
+    el = EventLog()
+    with pytest.raises(jsonschema.SchemaError):
+        el.register_schema({
+            # Totally invalid
+            'properties': True
+        })
+
+    with pytest.raises(ValueError):
+        el.register_schema({
+            'properties': {}
```

Comment:
Should this throw a `KeyError` instead? The problem is caused by missing keys (i.e. `$id` and `$version`), but I can't decide if that should result in a `ValueError` or `KeyError`.

Reply:
[KeyError](https://docs.python.org/3/library/exceptions.html#KeyError) says 'Raised when a mapping (dictionary) key is not found in the set of existing keys.', while [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) says 'Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.'

This makes me feel ValueError is the right thing, since we aren't passing anything in directly with the key.

Reply:
:+1: